js/dialogcards.js

```diff
@@ -853,6 +1088,126 @@ H5P.Dialogcards = (function ($, Audio, JoubelUI) {
     }
   };
 
+  /**
+   * Task is finished.
+   */
+
+  C.prototype.finishedScreen = function () {
+    var self = this;
+    self.taskFinished = true;
+    self.progress = -1;
+    self.triggerXAPIScored(1, 1, 'completed');
+
+    // Remove all these elements.
+    $('.h5p-dialogcards-cardwrap-set, .h5p-dialogcards-footer', self.$inner).remove();
+
+    // Display task finished feedback message.
+    $feedback = $('<div class="h5p-dialogcards-feedback">' + self.params.finished + '</div>').appendTo(self.$inner);
+
+    // Display reset button to enable user to do the task again.
+    self.$resetTaskButton = JoubelUI.createButton({
+      'class': 'h5p-dialogcards-reset',
+      'title': self.params.retry,
+      'html': self.params.retry
+    }).click(function () {
+      self.resetTask();
+    }).appendTo(self.$inner);
+  }
+
+
+  /**
+   * Remove card from DOM and from cards stack after user has checked the "gotit" button.
+   */
+
+  C.prototype.gotIt = function ($card) {
+    var self = this;
+    self.triggerXAPI('interacted');
+    var index = $card.index();
+
+    // Mark current card with a 'gotitdone' class.
+    self.$current.addClass('h5p-dialogcards-gotitdone');
+
+    // Move to next card if exists.
+    var $nextCard = self.$current.next('.h5p-dialogcards-cardwrap');
+    var $prevCard = self.$current.prev('.h5p-dialogcards-cardwrap');
+
+    if ($nextCard.length) {
+        var i = 1;
+        self.nextCard(i);
+    } else if ($prevCard.length) { // No next card left - go to previous.
+        self.prevCard();
+    } else { // No cards left: task is finished.
+        self.finishedScreen();
+        return;
+    }
+
+    // Now remove the current 'gotitdone' card from the cards and cardOrder arrays.
+      self.dialogs.splice(index, 1);
+      self.cardOrder.splice(index, 1);
+
+    // Remove the 'gotitdone' card from DOM
+      $( '.h5p-dialogcards-gotitdone', self.$inner).remove();
+
+      // Update navigation
+      self.updateNavigation();
+
+  };
+
+    /**
+   * Resets the task.
+   * Used in contracts.
+   * @public
+   */
+
+  C.prototype.resetTask = function () {
+    self = this;
+
+    // Removes all these elements to start afresh.
+    $('.h5p-dialogcards-cardwrap-set, .h5p-dialogcards-footer, .h5p-dialogcards-feedback, .h5p-dialogcards-reset, .h5p-dialogcards-order', self.$inner).remove();
```

Comment:
Perhaps using `self.$inner.html('')` or something similar is better. I'm guessing it will be hard to remember to add new elements to this list when changing code in the future.